### DISCUSSION TITLE
FYI, there is a critical low-level bug fixed in Silicon Labs EmberZNet 6.7.7 Zigbee stack

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#619] [General]
FYI, apparently, there is a very critical low-level bug fixed in Silicon Labs EmberZNet 6.7.7 stack and higher from Silabs:

https://community.openhab.org/t/zigbee-ikea-and-other-efr32mg1p-based-products-occasionally-crashing/109296/6

https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1261#issuecomment-739313960

Guess recommendation now is to upgrade Zigbee coordinator and devices to latest EmberZNet 6.7.x which currently is  6.7.8

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.7.8.0.pdf

This long-lived bug cause Silabs based products on the market to occasionally crash and no longer respond to APS frames.

**Comment by MattWestb** [comment]
I think the most ZHA devs already is knowing it and was verifying it in the deCONZ git isseu https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1261#issuecomment-739313960 and its very likely being in the code from the beginning of sirlabs zigbee PRO stack but its fixed in EZSP 6.7.7.0.
For the moment is nothing we can so before the manufacturer is releasing new firmware updates with the fix implanted. Or if some devs is finding one way rebooting routers that is being affected of the bug with one broadcast command.

All IKEA known OTA firmware is scanned for Zigbee Stack version and posted in the issue :-))))
IKEA have released 2 new OTA firmware for some day ago (The new LED drive and the new CWS3) that Zigbee stack version is for the moment not known.

The next but that is on the  NCP side and is breaking the config of pull control and the device is not going down in long pull mode is fixed in 6.7.8.0 https://github.com/zigpy/zigpy/discussions/595#discussioncomment-277760 and is very likely making the most problems with IKEA controlling devices (and also Samsung devices is reported in deCONZ and Philips HUE is reported in Tasmota) that is draining the batteries on Sonoff Zigbee Bridge. Looks its being very often triggered in second gens EFR32 and not so oft in the older radion chips.

The Tasmota devs is yesterday evening informed and is trying getting signed and unsigned firmware update from the factory for solving the problem.

**Comment by Hedda** [comment]
Anyone know which is the latest version of EmberZNet FW for IKEA Trådfri Signal Repeater E1746 and Tradfri Wireless Control Outlet?

That is, what version of EmberZNet is the latest official OTA firmware update from IKEA for the "E1746" and the "Outlet" is based on?

Follow-up question; does anyone happen to have any contacts at IKEA's Home Smart Business Unit so can ask them to update FW?

I can't even find the official method for contacting via mail or submitting written bug reports as IKEA of Sweden website's support section seem to just go in a loop when browsing their pages for customer service contact.

By the way, Björn Block looks to be the current head of the new IKEA Home Smart Business Unit at IKEA of Sweden, he has been working as the leader there since 2012 and his current title is Global Business Area Manager of IKEA Home Smart (and before that he was Global Business Leader of IKEA Home Smart and Global Range Manager of Lighting & IKEA Home smart):

https://newsroom.inter.ikea.com/news/ikea-invests-heavily-in-the-smart-home-going-forward/s/211c6cbc-6db5-4c5b-bb52-391741fca4c1

https://www.linkedin.com/in/bj%C3%B6rn-block-57753a30/

Though it should directly be in their interest it would probably too much to ask but could be very nice if developers for that department at IKEA would cooperate with projects like zigpy and zigbee-herdsman.

**Reply by MattWestb** [reply]
@puddly latest extraction of metadata is saying:

6.2.3.1    corrupted-current/10005777-6.1-TRADFRI-control-outlet-2.0.024.ota.ota.fw.bin
6.3.1.1    corrupted-current/10037603-3.1-TRADFRI-signal-repeater-2.2.005.ota.ota.fw.bin

Hi have not taking on look on the new CWS3 (the new color white spectrum GU10 that is released) and the new LED drive that is being released in the NL and perhaps some more market.

I have trying getting contact with some manager (Ivar S) of the department but there have not reacting.
The case is the 5 button remote and scene control groupe cant being bonded  and the same for the signal repeater.

**Comment by MattWestb** [comment]
By the way hi is well educated in Sydney and have working in Marbodal (Kitchen furitchers) and Ericsson.

If you like trying convert Björn to eng and use @ikea.com that looks being the normal syntax for their mail system.

Little dirty but can working but its on the border for stalking if not finding some official channel.

And its to late asking Ingvar :-((

Edit:  little more official but probably not working. You can calling and asking its one normal local telephone number.
pressoffice.rangesupply@inter.ikea.com
+46 732 321300

Or cal the switch board and asking kindly to talk with Mr Block it can working.

**Reply by Hedda** [reply]
As you say I'm not sure if that is OK to email him directly but yes his mail address should indeed be bjorn.block@ikea.com

https://www.mynewsdesk.com/se/ikea/images/bjoern-block-sortimentschef-foer-ikea-smart-home-894363

https://www.arbetslivsinstitutet.se/jobb/product-development-engineer/

I have only sent mails in English to ikeasverige@ikea.com and customer_care@ushelp.ikea.com with the above information.

**Reply by MattWestb** [reply]
They is having some more names that is on different certificate papirs they have applaying but I think its better getting the connection preity high in the organisation.

**Reply by Hedda** [reply]
FYI, Mario Flos and Daniel Lind both look to be Product Engineering Manager at IKEA Home Smart 

https://www.linkedin.com/in/marioflos/

https://www.linkedin.com/in/daniel-lind-4538bb44/

Another dirty move would be to search for "Product Development Engineer" OR "Product Design Engineer" AND "IKEA Home Smart"

**Reply by MattWestb** [reply]
The problem is that one ingegnere cant making the decision upgrading all firmware in their products then its costing very much time = much money.
The best world being one contact person that is accredited from the manager board for getting suggestions and then the board can taking on look on the "business case" and saying yes or now.

Wold being interesting getting 20 minutes talking with B Block and what hi is willing to invest for keeping or doing their position better on the "open community" part of the market that is not small part of there € in the end !!

I think they dont like doing the same mistake as OSRAM was doing and sutting ther self in both feets and killing the brand with bad products on the market.

**Comment by MattWestb** [comment]
My IKEA Billy EZSP is up and running on the fixed firmware :-)) :white_check_mark:
```
2021-01-15 15:10:01 INFO (MainThread) [bellows.zigbee.application] EZSP Radio manufacturer: IKEA of Sweden
2021-01-15 15:10:01 INFO (MainThread) [bellows.zigbee.application] EZSP Radio board name: Billy EZSP by MW
2021-01-15 15:10:01 INFO (MainThread) [bellows.zigbee.application] EmberZNet version: 6.7.8.0 build 373
```
The second one for testing and sniffing is also upgraded and also looks working well.

Tasmota have getting one updated "free" = unsigned firmware but ther devs have not replying if its working or not and i have not seen any request for one signed firmware (its primary one tasmota / sonoff problems not my personal then i was requesting the free one for them and also the first original signed one).
Edit: Unsigned and signed is in "release candidate" state. :black_square_button:
 
Its looks like "someone" have cooked Elelabs-ELU013 / ELR023 and EByte-E180-Z120B and starting to testing them. :black_square_button:

EM357 is cooked, verified and working OK. :white_check_mark:

And EM3581 is also getting one try but its little more tricky then the dev dont have one reference hardware for testing. Is cooked but not verified to working. :black_square_button:

Do i have forgetting lobbying for some device that need one EZSP 6.7.8.0 upgrade :interrobang:

I holding my finger crossed that all is going to working well and its being better for the ZHA community to getting one "golden release" that is possible to upgrading our coordinators with, and can drawing one line over old EZSP bugs that making our devs busy hunting and coding work arounds for, instead of putting efforts in coding new nice futures in our systems.
